implementations/micrometer-registry-signalfx/src/test/java/io/micrometer/signalfx/SignalFxNamingConventionTest.java

```diff
@@ -30,4 +36,16 @@ void tagKey() {
         assertThat(convention.tagKey("123")).isEqualTo("a123");
     }
 
-}
\ No newline at end of file
+    @Test
+    void tagKeyWhenKeyHasBlacklistedCharShouldSanitize() {
+        assertThat(convention.tagKey("a.b")).isEqualTo("a_b");
+    }
+
+    @Test
+    void tagKeyWhenKeyStartsWithBlacklistedPrefixShouldSanitize() {
+        assertThat(convention.tagKey("aws_a")).isEqualTo("aaws_a");
+        assertThat(convention.tagKey("gcp_a")).isEqualTo("agcp_a");
+        assertThat(convention.tagKey("azure_a")).isEqualTo("aazure_a");
```

Comment:
This isn't ideal. I feel like it'd be better if we logged a warning the first time to let users know what metric is registered that violates the rules of the metrics system, and then just don't send the metric at all. I don't think we have a proper way of doing that last part now, though. Maybe in the future we need a distinction between converting for convention and violations with no obvious conversion. It's subjective, though.
Anyways, for now, what does everyone think of adding a once logger to warn of the metric name and otherwise keep things as they are?

Reply:
@shakuzen Thanks for the feedback! I just chose this way as leading underscores are already replaced with "a". I changed it to use the same approach as `DynatraceNamingConvention` in 813505a. Do you think that leading underscores need to be treated in the same way?

Reply:
Leading underscores and numbers somehow felt more acceptable to me the way they are now, though the argument could be made for them too. The next question is, if we're not changing these now and attempt to send them as is, does the whole payload fail or are only these metrics not stored by signalfx? We don't want the whole payload to fail because some don't follow the rules.

Reply:
If the whole payload fails, it'd be better to restore the behavior of appending the `a` character - at least it is consistent with the other cases you mentioned, and at least we're logging something now so it can be noticed and fixed.

Reply:
> The next question is, if we're not changing these now and attempt to send them as is, does the whole payload fail or are only these metrics not stored by signalfx?

@shakuzen I'm not a SignalFx user, so I'm not sure. I just read the following description from @kjurasovic-dexcom in #1746 as the other metrics haven't been rejected:

```
The problem is SignalFx rejects all Kafka related metrics.
```

Reply:
@izeye from the tests we did it seems like SignalFx only rejects metrics that don't follow the rules. Other JVM metrics were displayed correctly in SignalFx. I could check with them directly if needed?

Reply:
@kjurasovic-dexcom Thanks for the feedback! If you could confirm it, it would be much appreciated.

Reply:
@izeye This is the info I got from SignalFx team "we’ll just drop those metrics that have this invalid tag name.  Others should be ok.". This is the behavior we observed in our tests.

Reply:
@kjurasovic-dexcom Thanks for the confirmation!

@shakuzen Based on the info, is it okay with as-is?

Reply:
Looks good to me. Thank you both for the follow-up on this.